packages/client-config/src/paths/get_client_config_path.ts

```diff
@@ -23,11 +25,18 @@ export const getClientConfigPath = async (
   let targetPath = defaultArgs.out;
 
   if (outDir) {
-    const outDirIsFile = fs.lstatSync(outDir).isFile();
-    if (!outDirIsFile) {
-      targetPath = path.isAbsolute(outDir)
-        ? outDir
-        : path.resolve(process.cwd(), outDir);
+    targetPath = path.isAbsolute(outDir)
+      ? outDir
+      : path.resolve(process.cwd(), outDir);
+
+    const relativeTargetPath = path.relative(process.cwd(), targetPath);
+    try {
+      debug?.(`Checking if directory ${relativeTargetPath} exists...`);
+      await fsp.access(outDir);
+    } catch (error) {
+      // outDir does not exist, so create dir
+      await fsp.mkdir(outDir, { recursive: true });
+      debug?.(`Directory created: ${relativeTargetPath}`);
     }
```

Comment:
FS access is fast. These messages are going to be printed at almost exact same time. Perhaps there should only be one message?
Or perhaps we could just not log anything (to somewhat solve abstraction problem by eliminating the need) ?

Reply:
I'll double check and see if these messages are needed since the original log `File written: ...` contains the directory created anyways.

Reply:
Removed the debug messages as there is not much benefit on top of the original `File written: ...` message.